app/main.py

```diff
@@ -1,6 +1,21 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cached = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, tuple(kwargs.items()))
+        if key in cached:
+            print("Getting from cache")
+            return cached[key]
+
+        print("Calculating new result")
+        result = func(*args, **kwargs)
+        cached[key] = result
+        return result
```

Comment:
try to rewrite your decorator not to have `return` duplicated

Reply:
got it, commited fix